flows/engine/engine.go

```diff
@@ -143,24 +139,30 @@ func resumeNode(run flows.FlowRun, node flows.Node, step flows.Step, event flows
 		return noDestination, step, nil
 	}
 
-	err := wait.End(run, step, event)
+	// try to resume our wait with the first caller event
+	err := wait.End(run, step, callerEvents[0])
 	if err != nil {
 		run.AddError(step, err)
 		run.Exit(flows.StatusErrored)
 		return noDestination, step, nil
 	}
 
+	// we can now apply the caller events
+	for _, event := range callerEvents {
+		run.ApplyEvent(step, event)
+	}
```

Comment:
The wait is the gate keeper than decides if flow execution can proceed - so I think it makes sense that caller events are applied after that. I'll document that somewhere..